src/main/scala/swiss/dasch/domain/BulkIngestService.scala

```diff
@@ -38,22 +38,26 @@ final case class BulkIngestService(
       _         <- semaphoresPerProject.put(key, semaphore)
     } yield semaphore).commit
 
-  private def acquireWithTimeout(sem: TSemaphore): UIO[Unit] =
-    sem.acquire.commit.timeout(Duration.fromMillis(400)).as(())
+  private def acquireWithTimeout(sem: TSemaphore): IO[Unit, Unit] =
+    sem.acquire.commit.timeout(Duration.fromMillis(400)).some.mapError(_ => ())
 
-  private def withSemaphore[E, A](key: ProjectShortcode)(
+  private def withSemaphoreDaemon[E, A](key: ProjectShortcode)(
     zio: IO[E, A],
-  ): IO[Option[E], A] =
+  ): IO[Unit, Fiber.Runtime[E, A]] =
     getSemaphore(key)
       .tap(acquireWithTimeout(_).asSomeError)
```

Comment:
*idea* Just a minor improvement, reducing the times .commit is called:

Come to think of it: Instead of having two different transactions we could have a single, if you remove the `commit` in the `getSemaphore`:
`(getSemaphore(key).flatMap(sem => sem.acquire.as(sem))).commit.timeout(400.millis).some.orElseFail…`

Reply:
👍  Single commit works, see the change.